packages/flutter/lib/src/widgets/overscroll_indicator.dart

```diff
@@ -637,18 +644,285 @@ class _GlowingOverscrollIndicatorPainter extends CustomPainter {
+/// [OverscrollIndicatorNotification] before showing an overscroll indication.
+/// To prevent the indicator from showing the indication, call
+/// [OverscrollIndicatorNotification.disallowStretch] on the notification.
+///
+/// Created by [ScrollBehavior.buildOverscrollIndicator] on platforms
+/// (e.g., Android) that commonly use this type of overscroll indication when
+/// [ScrollBehavior.updatedAndroidOverscrollIndicator] is true. Otherwise, the
+/// default [GlowingOverscrollIndicator] is applied.
 ///
 /// See also:
 ///
-///  * [GlowingOverscrollIndicator], which generates this type of notification.
+///  * [OverscrollIndicatorNotification], which can be used to prevent the stretch
+///    effect from being applied at all.
+///  * [NotificationListener], to listen for the
+///    [OverscrollIndicatorNotification]
+class StretchingOverscrollIndicator extends StatefulWidget {
+  /// Creates a visual indication that a scroll view has overscrolled by
+  /// applying a stretch transformation to the content.
+  ///
+  /// In order for this widget to display an overscroll indication, the [child]
+  /// widget must contain a widget that generates a [ScrollNotification], such
+  /// as a [ListView] or a [GridView].
+  ///
+  /// The [axisDirection] and [notificationPredicate] arguments must not be null.
+  const StretchingOverscrollIndicator({
+    Key? key,
+    required this.axisDirection,
+    this.notificationPredicate = defaultScrollNotificationPredicate,
+    this.child,
+  }) : assert(axisDirection != null),
+        assert(notificationPredicate != null),
+        super(key: key);
+
+  /// {@macro flutter.overscroll.axisDirection}
+  final AxisDirection axisDirection;
+
+  /// {@macro flutter.overscroll.axis}
+  Axis get axis => axisDirectionToAxis(axisDirection);
+
+  /// {@macro flutter.overscroll.notificationPredicate}
+  final ScrollNotificationPredicate notificationPredicate;
+
+  /// The widget below this widget in the tree.
+  ///
+  /// The overscroll indicator will apply a stretch effect to this child. This
+  /// child (and its subtree) should include a source of [ScrollNotification]
+  /// notifications.
+  ///
+  /// Typically a [StretchingOverscrollIndicator] is created by a
+  /// [ScrollBehavior.buildOverscrollIndicator] method when opted-in using the
+  /// [ScrollBehavior.updatedAndroidOverscrollIndicator] flag. In this case
+  /// the child is usually the one provided as an argument to that method.
+  final Widget? child;
+
+  @override
+  State<StretchingOverscrollIndicator> createState() => _StretchingOverscrollIndicatorState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(EnumProperty<AxisDirection>('axisDirection', axisDirection));
+  }
+}
+
+class _StretchingOverscrollIndicatorState extends State<StretchingOverscrollIndicator> with TickerProviderStateMixin {
+  _StretchController? _stretchController;
```

Comment:
```suggestion
  late final _StretchController _stretchController = _StretchController(vsync: this);
```

and then remove the `initState` and a whole bunch of `!`.